qutebrowser/completion/completer.py

```diff
@@ -121,12 +114,11 @@ def _open_completion_if_needed(self):
         if not config.get('completion', 'auto-open'):
             connected = self._connect_signals(True)
             if connected:
-                self.update_completion()
+                self._update_completion()
 
     def _model(self):
         """Convenience method to get the current completion model."""
-        completion = objreg.get('completion', scope='window',
-                                window=self._win_id)
+        completion = self.parent()
```

Comment:
I kind of avoided doing this in the past and set an explicit attribute (with another argument to `__init__`, even if it means I'm mentioning the parent twice), because it could be a pitfall when you need to change the parent for some reason - you wouldn't necessarily realize the consequences.

However, I don't really have a strong opinion on this - what do you think?